swiperefresh/src/main/java/com/google/accompanist/swiperefresh/SwipeRefreshIndicator.kt

```diff
@@ -0,0 +1,199 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import kotlin.math.min
+
+/**
+ * Object which contains default values for [SwipeRefreshIndicator].
+ */
+object SwipeRefreshIndicatorDefaults {
+    /**
+     * The default/normal size values for [SwipeRefreshIndicator].
+     */
+    val DefaultSizes = SwipeRefreshIndicatorSizes(
+        size = 40.dp,
+        arcRadius = 7.5.dp,
+        strokeWidth = 2.5.dp,
+        arrowWidth = 10.dp,
+        arrowHeight = 5.dp,
+    )
+
+    /**
+     * The 'large' size values for [SwipeRefreshIndicator].
+     */
+    val LargeSizes = SwipeRefreshIndicatorSizes(
+        size = 56.dp,
+        arcRadius = 11.dp,
+        strokeWidth = 3.dp,
+        arrowWidth = 12.dp,
+        arrowHeight = 6.dp,
+    )
+}
+
+/**
+ * A class to encapsulate details of different indicator sizes.
+ *
+ * @param size The overall size of the indicator.
+ * @param arcRadius The radius of the arc.
+ * @param strokeWidth The width of the arc stroke.
+ * @param arrowWidth The width of the arrow.
+ * @param arrowHeight The height of the arrow.
+ */
+@Immutable
+data class SwipeRefreshIndicatorSizes(
+    val size: Dp,
+    val arcRadius: Dp,
+    val strokeWidth: Dp,
+    val arrowWidth: Dp,
+    val arrowHeight: Dp,
+)
+
+/**
+ * Indicator composable which is typically used in conjunction with [SwipeRefresh].
+ *
+ * @param isRefreshing Whether the indicator should display an indeterminate progress indicator.
+ * @param offset The current scroll offset, in pixels.
+ * @param triggerOffset The scroll offset which would trigger a refresh, in pixels.
+ * @param modifier The modifier to apply to this layout.
+ * @param scale Whether the indicator should scale up/down as it is scrolled in. Defaults to false.
+ * @param arrowEnabled Whether an arrow should be drawn on the indicator. Defaults to true.
+ * @param backgroundColor The color of the indicator background surface.
+ * @param contentColor The color for the indicator's contents.
+ * @param shape The shape of the indicator background surface. Defaults to [CircleShape].
+ * @param sizes The [SwipeRefreshIndicatorSizes] to use. Defaults to
+ * [SwipeRefreshIndicatorDefaults.DefaultSizes].
+ * @param elevation The size of the shadow below the indicator.
+ */
+@Composable
+fun SwipeRefreshIndicator(
+    isRefreshing: Boolean,
+    offset: Float,
+    triggerOffset: Float,
+    modifier: Modifier = Modifier,
+    scale: Boolean = false,
+    arrowEnabled: Boolean = true,
+    backgroundColor: Color = MaterialTheme.colors.surface,
+    contentColor: Color = contentColorFor(backgroundColor),
+    shape: CornerBasedShape = CircleShape,
+    sizes: SwipeRefreshIndicatorSizes = SwipeRefreshIndicatorDefaults.DefaultSizes,
```

Comment:
I don't mind having this data class, but I'm curious how good for UX to allow such customisations? It might be that just a `largeIndication: Boolean = false` is enough and if people really want a fancier arc -- they could make their own indicator. 

The reason I suggest that is that I want those indicators to feel/look the same in other apps, so we could make it a but more complex to go out of the box, but still support large/small spec distinctions.

Curious what you think about it

Reply:
Yeah agreed. Added a `largeIndication` param, and hid the data class for now.